cdap-system-app-api/src/main/java/io/cdap/cdap/api/service/worker/RunnableTaskRequest.java

```diff
@@ -31,13 +31,17 @@
   private final ArtifactId artifactId;
   @Nullable
   private final String namespace;
+  @Nullable
+  private final String wrappedClassName;
 
   private RunnableTaskRequest(String className, @Nullable String param,
-                              @Nullable ArtifactId artifactId, @Nullable String namespace) {
+                              @Nullable ArtifactId artifactId, @Nullable String namespace,
+                              @Nullable String wrappedClassName) {
     this.className = className;
     this.param = param;
     this.artifactId = artifactId;
     this.namespace = namespace;
+    this.wrappedClassName = wrappedClassName;
```

Comment:
Have a field like this is awkward. The information is already in the `param`. We should use a class for the `param` instead of `String` such that structure can be enforced better.

Reply:
Done.